pkg/nuctl/test/function_test.go

```diff
@@ -689,6 +691,146 @@ func (suite *functionGetTestSuite) TestDelete() {
 	suite.Require().NoError(err, "Function was suppose to be deleted!")
 }
 
+type functionExportImportTestSuite struct {
+	Suite
+}
+
+func (suite *functionExportImportTestSuite) TestExportImportRoundTrip() {
+	uniqueSuffix := "-" + xid.New().String()
+	functionName := "reverser" + uniqueSuffix
+	imageName := "nuclio/processor-" + functionName
+
+	namedArgs := map[string]string{
+		"path":    path.Join(suite.GetFunctionsDir(), "common", "reverser", "golang"),
+		"runtime": "golang",
+		"handler": "main:Reverse",
+	}
+
+	err := suite.ExecuteNuctl([]string{"deploy", functionName, "--verbose", "--no-pull"}, namedArgs)
+	suite.Require().NoError(err)
+
+	// make sure to clean up after the test
+	defer suite.dockerClient.RemoveImage(imageName)
+
+	// use nutctl to delete the function when we're done
+	defer suite.ExecuteNuctl([]string{"delete", "fu", functionName}, nil)
+
+	// export the function
+	err = suite.ExecuteNuctlAndWait([]string{"export", "fu", functionName}, map[string]string{}, false)
+	suite.Require().NoError(err)
+
+	exportedFunctionConfig := functionconfig.Config{}
+	err = yaml.Unmarshal(suite.outputBuffer.Bytes(), &exportedFunctionConfig)
+	suite.Require().NoError(err)
+
+	// assert skip annotations
+	suite.Assert().True(functionconfig.ShouldSkipBuild(exportedFunctionConfig.Meta.Annotations))
+	suite.Assert().True(functionconfig.ShouldSkipDeploy(exportedFunctionConfig.Meta.Annotations))
+
+	// edit the exported function's name
+	exportedFunctionName := "export-reverser-" + xid.New().String()
+	exportedFunctionConfig.Meta.Name = exportedFunctionName
+	exportedFunctionConfigJson, _ := json.Marshal(exportedFunctionConfig)
+
+	// write exported function config to temp file
+	exportTempFile, err := ioutil.TempFile("", "reverser.*.json")
+	suite.Require().NoError(err)
+	defer os.Remove(exportTempFile.Name())
+
+	_, err = exportTempFile.Write(exportedFunctionConfigJson)
+	suite.Require().NoError(err)
+
+	// import the function
+	err = suite.ExecuteNuctl([]string{"import", "fu", exportTempFile.Name()}, map[string]string{})
+	suite.Require().NoError(err)
+
+	// use nutctl to delete the function when we're done
+	defer suite.ExecuteNuctl([]string{"delete", "fu", exportedFunctionName}, nil)
+
+	// wait until able to get the function
+	err = suite.ExecuteNuctlAndWait([]string{"get", "function", exportedFunctionName}, map[string]string{}, false)
+	suite.Require().NoError(err)
+
+	// deploy imported function
+	err = suite.ExecuteNuctl([]string{"deploy", exportedFunctionName, "--verbose"}, map[string]string{})
```

Comment:
I would add an "invoke" command on function and expect to fail, to ensure it wasn't imported + deployed